src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+from azure.cli.core.commands.parameters import name_type, tags_type, get_three_state_flag, get_enum_type
+
+
+# pylint: disable=too-many-statements
+def load_arguments(self, _):
+    # synapse spark
+
+    for scope in ['batch', 'session', 'session-statement']:
+        with self.argument_context('synapse spark ' + scope) as c:
+            c.argument('workspace_name', help='The name of the workspace.')
+            c.argument('spark_pool_name', help='The name of the spark pool.')
+
+    for scope in ['synapse spark batch', 'synapse spark session']:
+        with self.argument_context(scope + ' create') as c:
+            c.argument('artifact_id', help='The artifact id.')
+            c.argument('job_name', arg_type=name_type, help='The spark batch or session job name.')
+            c.argument('file', help='The URI of file.')
+            c.argument('class_name', help='The class name.')
+            c.argument('args', nargs='+', help='The arguments of the job.')
+            c.argument('jars', nargs='+', help='The array of jar files.')
+            c.argument('files', nargs='+', help='The array of files URI.')
+            c.argument('archives', nargs='+', help='The array of archives.')
+            c.argument('conf', help='The configuration of spark batch job.')
+            c.argument('driver_memory', help='The memory of driver.')
+            c.argument('driver_cores', help='The number of cores in driver.')
+            c.argument('executor_memory', help='The memory of executor.')
+            c.argument('executor_cores', help='The number of cores in each executor.')
+            c.argument('num_executors', help='The number of executors.')
+            c.argument('tags', arg_type=tags_type, help='The tags of spark batch job.')
+            c.argument('detailed', arg_type=get_three_state_flag(),
+                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+
+        with self.argument_context(scope + ' list') as c:
+            c.argument('from_index', help='Optional parameter specifying which index the list should begin from.')
+            c.argument('detailed', arg_type=get_three_state_flag(),
+                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+            c.argument('size',
+                       help='The size of the returned list.By default it is 20 and that is the maximum.')
+
+    with self.argument_context('synapse spark batch create') as c:
+        c.argument('job_type', arg_type=get_enum_type(['SCALA', 'PYTHON', 'DOTNET'], default='SCALA'),

```

Comment:
>job_type [](start = 20, length = 8)

I would suggest to rename the parameter to `language` to keep consistency with Synapse Studio. `job_type` is kind of inaccurate, users might think it as something like "Spark Batch job" or "spark session job"

Reply:
> > job_type [](start = 20, length = 8)
> 
> I would suggest to rename the parameter to `language` to keep consistency with Synapse Studio. `job_type` is kind of inaccurate, users might think it as something like "Spark Batch job" or "spark session job"

fixed